File: two-sum/two-sum.js
const twoSum = function(nums, target) {
    const newSet = new Map();
    for(let i=0; i<nums.length; i++) {
        newSet.set(nums[i],i);
    }
    
    const indexes = [];
    for(let i=0; i<nums.length; i++) {
        const findKey = target - nums[i];
        if(newSet.get(findKey) != i && newSet.has(findKey)) {
            indexes.push(i);
            indexes.push(newSet.get(findKey));
            break;
        }
    }
    return indexes;
};

File: Arrays/two-sum/two-sum.cpp
//complexity - O(n)

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map<int,int> mp;
        vector<int> ans;
        for(int i=0;i<n;i++) mp[nums[i]] = i;
        for(int i=0;i<n;i++){
            int d = target - nums[i];
            int index = mp[d];
            if(nums[index]==d and index>i){
               ans.push_back(i);
                ans.push_back(index);
                break;
            }
        }
        return ans;
    }
};
